s3po/backends/swift.py

```diff
@@ -43,8 +43,8 @@ def func():
                     raise DownloadException('Downloaded only %i of %i bytes' % (
                         fobj.count, int(length)))
 
-            except ClientException:
-                raise DownloadException('Key not found.')
+            except ClientException as exc:
+                raise DownloadException('Key not found: ' + str(exc))
```

Comment:
If we were using Python 3, this could be a [chained exception](http://stackoverflow.com/a/16414892/1307166).